From 5e4278d12c5bf8c40b7c0b431fda4f67b133cfc9 Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Fri, 17 Oct 2014 01:06:38 +0200
Subject: [PATCH 5/9] ArmPlatformPkg/Bds: initialize ConIn/ConOut/ErrOut before
 connecting terminals

In the following call tree:

  BdsEntry()
    DefineDefaultBootEntries()
      BdsConnectAllDrivers()
    InitializeConsole()
      set ConIn/ConOut/ErrOut

BdsConnectAllDrivers() connects SerialDxe -> TerminalDxe -> ConPlatformDxe
-> ConSplitterDxe before InitializeConsole has a chance to set ConIn /
ConOut / ErrOut. This causes ConPlatformDxe, at very first boot, to filter
out TerminalDxe's STI and STO from the set that ConSplitterDxe
multiplexes, leaving the system without a terminal console.

Reorder InitializeConsole() with DefineDefaultBootEntries(), so that the
variables be always set when DefineDefaultBootEntries() calls
BdsConnectAllDrivers().

Contributed-under: TianoCore Contribution Agreement 1.0
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
---
 ArmPlatformPkg/Bds/Bds.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ArmPlatformPkg/Bds/Bds.c b/ArmPlatformPkg/Bds/Bds.c
index b197c96..3ad5e2e 100644
--- a/ArmPlatformPkg/Bds/Bds.c
+++ b/ArmPlatformPkg/Bds/Bds.c
@@ -620,9 +620,6 @@ BdsEntry (
         0, NULL);
   }
 
-  // If Boot Order does not exist then create a default entry
-  DefineDefaultBootEntries ();
-
   // Now we need to setup the EFI System Table with information about the console devices.
   InitializeConsole ();
 
@@ -633,6 +630,9 @@ BdsEntry (
   Status = gBS->CalculateCrc32 ((VOID*)gST, gST->Hdr.HeaderSize, &gST->Hdr.CRC32);
   ASSERT_EFI_ERROR (Status);
 
+  // If Boot Order does not exist then create a default entry
+  DefineDefaultBootEntries ();
+
   // Timer before initiating the default boot selection
   StartDefaultBootOnTimeout ();
 
-- 
1.8.3.1

